akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala

```diff
@@ -176,6 +173,11 @@ object EventSourcedBehavior {
    */
   def withSnapshotSelectionCriteria(selection: SnapshotSelectionCriteria): EventSourcedBehavior[Command, Event, State]
 
+  /**
+    * Criteria for internal retention/deletion of snapshots and events.
+    */
+  def withRetention(criteria: RetentionCriteria): EventSourcedBehavior[Command, Event, State]
+
```

Comment:
corresponding is missing in javadsl, there it should be a method in `EventSourcedBehavior` that can be overridden:
```
def retentionCriteria: RetentionCriteria = RetentionCriteria()
```